src/shared/components/lollipopMutationPlot/LollipopMutationPlot.tsx

```diff
@@ -173,65 +179,76 @@ export default class LollipopMutationPlot extends React.Component<ILollipopMutat
                 <div>
                     <a
-                        href={`http://www.uniprot.org/uniprot/${this.props.store.uniprotId.result}`}
-                        target="_blank"
-                    >
-                        {this.props.store.uniprotId.result}
-                    </a>
-                    <a
-                        style={{marginLeft:"5px"}}
+                        style={{marginRight:"5px"}}
                         href={`http://pfam.xfam.org/family/${pfamAccession}`}
                         target="_blank"
                     >
                         PFAM
                     </a>
-                </div>
-                <div>
                     {mutationAlignerA}
                 </div>
             </div>
         );
     }
 
     @computed private get domains():DomainSpec[] {
-        if (!this.props.store.pfamGeneData.isComplete || !this.props.store.pfamGeneData.result.regions) {
+        if (!this.props.store.pfamDomainData.isComplete ||
+            !this.props.store.pfamDomainData.result ||
+            this.props.store.pfamDomainData.result.length === 0)
+        {
             return [];
         } else {
-            return this.props.store.pfamGeneData.result.regions.map((region:any)=>{
-                const startCodon:number = region.metadata.start;
-                const endCodon:number = region.metadata.end;
-                const label:string = region.metadata.identifier;
-                const color:string = region.colour;
-                const tooltip:JSX.Element = this.domainTooltip(region);
+            return this.props.store.pfamDomainData.result.map((domain:PfamDomain)=>{
                 return {
-                    startCodon,
-                    endCodon,
-                    label,
-                    color,
-                    tooltip
+                    startCodon: domain.pfamDomainStart,
+                    endCodon: domain.pfamDomainEnd,
+                    label: domain.pfamDomainName,
+                    color: this.domainColorMap[domain.pfamDomainId],
+                    tooltip: this.domainTooltip(domain)
                 };
             });
         }
     }
 
+    @computed private get domainColorMap(): {[pfamAccession:string]: string}
+    {
+        const map:{[pfamAccession:string]: string} = {};
+        const colors: string[] = ["#2dcf00", "#ff5353", "#5b5bff", "#ebd61d", "#ba21e0",
+            "#ff9c42", "#ff7dff", "#b9264f", "#baba21", "#c48484",
+            "#1f88a7", "#cafeb8", "#4a9586", "#ceb86c", "#0e180f"];
+        let colorIdx = 0;
+
+        if (!this.props.store.pfamDomainData.isPending &&
+            this.props.store.pfamDomainData.result)
+        {
+            this.props.store.pfamDomainData.result.sort(
+                (a: PfamDomain, b: PfamDomain): number =>
+                    a.pfamDomainStart - b.pfamDomainStart
+            ).forEach((domain:PfamDomain) => {
+                if (map[domain.pfamDomainId] === undefined)
```

Comment:
Assigning a different color for each unique domain starting from the beginning (starting from the domain with the smaller start position)